ui/src/main/java/org/pentaho/di/ui/spoon/tree/extension/SashResizer.java

```diff
@@ -0,0 +1,171 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2024 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.ui.spoon.tree.extension;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles simplified resizing weights for a SashForm, storing last known sizes and normalizing them to 100.
+ */
+public class SashResizer {
+  private static Logger log = LoggerFactory.getLogger( SashResizer.class );
+
+  private static final int NORM_TARGET = 100;
+  private static final int DEFAULT_WEIGHT = 30;
+
+  /** list of last non-zero percentage sizes recorded for elements (can add up to more than 100) */
+  private int[] weights;
+  private boolean[] enabled;
```

Comment:
I wonder if we should store these so they are preserved on restart.

Reply:
enabled state needs to be determined by the plugin, no guarantee it will stay the same after restart

Reply:
maybe the weights? Seems like Spoon remembers the size of most things.